node/uhpo/src/payout_update.rs

```diff
@@ -0,0 +1,275 @@
+
+        let data: [u8; 32] = rng.gen();
+        let keypair = SecretKey::from_slice(&data).unwrap().keypair(&secp);
+
+        let new_payout_address: Address =
+            Address::p2tr(&secp, keypair.x_only_public_key().0, None, Network::Regtest);
+
+        (secp, keypair, new_payout_address)
+    }
+
+    #[test]
+    fn test_new_payout_update_only_cb() {
+        let (_, _, new_payout_address) = setup();
+
+        let coinbase_tx = create_dummy_transaction();
+        let prev_update_tx = None;
+        let projected_fee = Amount::from_sat(1000);
+
+        let payout_update = PayoutUpdate::new(
+            prev_update_tx,
+            coinbase_tx,
+            new_payout_address,
+            projected_fee,
+        )
+        .expect("Failed to create payout update");
+
+        assert_eq!(payout_update.transaction.input.len(), 1);
+        assert_eq!(payout_update.transaction.output.len(), 1);
+        assert_eq!(
+            payout_update.transaction.output[0].value,
+            Amount::from_sat(50000000) - projected_fee
+        );
+        assert!(payout_update.prev_update_txout.is_none());
+    }
+
+    #[test]
+    fn test_new_payout_update() {
+        let (_, _, new_payout_address) = setup();
+
+        let coinbase_tx = create_dummy_transaction();
+        let prev_update_tx = create_dummy_transaction();
+        let projected_fee = Amount::from_sat(1000);
+
+        let payout_update = PayoutUpdate::new(
+            Some(prev_update_tx),
+            coinbase_tx,
+            new_payout_address,
+            projected_fee,
+        )
+        .expect("Failed to create payout update");
+
+        assert_eq!(payout_update.transaction.input.len(), 2);
+        assert_eq!(payout_update.transaction.output.len(), 1);
+        assert_eq!(
+            payout_update.transaction.output[0].value,
+            Amount::from_sat(50000000 * 2) - projected_fee
+        );
+        assert!(payout_update.prev_update_txout.is_some());
+    }
+
+    #[test]
+    fn test_add_signature() {
+        let (_, keypair, new_payout_address) = setup();
+
+        let coinbase_tx = create_dummy_transaction();
+        let prev_update_tx = create_dummy_transaction();
+        let projected_fee = Amount::from_sat(1000);
+        let mut payout_update = PayoutUpdate::new(
+            Some(prev_update_tx),
+            coinbase_tx,
+            new_payout_address,
+            projected_fee,
+        )
+        .expect("Failed to create payout update");
+
+        payout_update
+            .add_coinbase_sig(&keypair.secret_key(), &None)
+            .expect("Failed to add coinbase signature");
+
+        payout_update
+            .add_prev_update_sig(&keypair.secret_key(), &None)
+            .expect("Failed to add prev update signature");
+
+        assert_eq!(payout_update.transaction.input[0].witness.len(), 1);
+        assert_eq!(payout_update.transaction.input[1].witness.len(), 1);
+    }
+
+    fn create_dummy_transaction() -> Transaction {
```

Comment:
Move this function to the top, below the setup. All support functions should come before the test functions.